### ISSUE TITLE
Cannot construct `torch.sparse_coo_tensor` (but `scipy.sparse.coo_matrix` works fine): `TypeError: only integer tensors of a single element can be converted to an index`

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#109108]
### 🐛 Describe the bug

Also, note the strange warning when using `weights_only=True`.
[bug.pt.gz](https://github.com/pytorch/pytorch/files/12586855/bug.pt.gz)

```python
import gzip
import torch
import scipy

bug = torch.load(gzip.open('bug.pt.gz', 'rb'), weights_only=True)
#  /home/vadimkantorov/.local/lib/python3.8/site-packages/torch/_utils.py:841: UserWarning: TypedStorage is deprecated. It will be removed in the future and UntypedStorage will be the only storage class. This should only matter to you if you are using storages directly.  To access UntypedStorage directly, use tensor.untyped_storage() instead of tensor.storage()
#  return self.fget.__get__(instance, owner)()

scipy.sparse.coo_matrix((bug['values'], (bug['rowinds'], bug['colinds'])), shape=bug['shape'])
# ok

torch.sparse_coo_tensor((bug['rowinds'], bug['colinds']), bug['values'], size=bug['shape'])
# TypeError: only integer tensors of a single element can be converted to an index
```

Same error appears in https://github.com/pytorch/pytorch/issues/43949 in a completely different context. This error is extremely mysterious and weird.

### Versions

2.1.0.dev20230802+cpu

**Comment by vadimkantorov** [comment]
Related on args inconsistancies compared to scipy:
- 
-https://github.com/pytorch/pytorch/issues/109017

**Comment by pearu** [comment]
I think this is a documentation issue. The `indices` can be tuple or other types but not of mixed types. For example,
```python
torch.sparse_coo_tensor((tuple(bug['rowinds']), tuple(bug['colinds'])), bug['values'], size=bug['shape'])
```
works fine, although, don't use it as it is very inefficent. Using the torch.stack as in https://github.com/pytorch/pytorch/issues/109108#issuecomment-1715735709 is the recommended approach.

**Comment by vadimkantorov** [comment]
I think it makes sense to accept rowind/colinds as 1d tensor without forcing to stack (and without converting these huge arrays of numbers to Python objects) - same as scipy supports it

**Comment by pearu** [comment]
Internally, in Torch, `indices` is a two-dimensional tensor. So, without changing related internals in torch sparse support, somewhere the `torch.stack` needs to be used. Currently, it is the user's responsibility.

I haven't checked how scipy supports it but I suspect it considers `indices` as an iterator of iterators. In the context of Torch, the issue is about supporting the construction of a tensor from an iterator of iterators. Currently, this is unsupported in torch:
```python
>>> torch.tensor((torch.tensor([1, 2]), torch.tensor([3, 4])))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: only integer tensors of a single element can be converted to an index
```

**Comment by pearu** [comment]
> and without converting these huge arrays of numbers to Python objects

When using `torch.stack`, the conversion to Python objects never happens.

**Comment by vadimkantorov** [comment]
> the conversion to Python objects never happens.

I was referring to your example of using `tuple(bug['colinds'])`

Overall, I agree, but the differences with NumPy should be clearly explained in docs and even better if these are clearly explained in the error messages as well.

**Comment by vadimkantorov** [comment]
> I haven't checked how scipy supports it but I suspect it considers `indices` as an iterator of iterators. In the context of Torch, the issue is about supporting the construction of a tensor from an iterator of iterators. Currently, this is unsupported in torch:

Ha-ha, I even happen to have filed an issue about this (in another context): 
- https://github.com/pytorch/pytorch/issues/71479

**Comment by amjames** [comment]
@vadimkantorov I agree that documenting the divergence from scipy/numpy api in these cases is the correct course to take. 

If someone wanted to submit a docs change feel free to ping me for review.

**Comment by kit1980** [comment]
I've got some very early attempt at the warning, but it's in progress https://github.com/pytorch/pytorch/pull/113498
